### DISCUSSION TITLE
If a Processor Plugin returns an empty message is the message dropped?

### DISCUSSION THREAD

**Discussion by Degiorgio** [redpanda-data/connect#2342] [General]
Hello,

If I have Processor plugin, thats returns an empty `service.MessageBatch`, something like this:

```
func (p *someplugin) Process(_ context.Context, m *service.Message) (service.MessageBatch, error) {
         ....
	return []*service.Message{}, nil
}
```
Is it safe to assume that the message is dropped? (i.e. processors downstream of this processor won't be called?)

Another question I have, is the constructor of processor plugin called once or per message?

**Comment by mihaitodor** [comment]
Hey! 👋 

> Is it safe to assume that the message is dropped? (i.e. processors downstream of this processor won't be called?)

Yes, any message which passes through that processor will be deleted (just like using a `mapping` processor with `root = deleted()`) and an ack will be sent to the input if the input supports acknowledgements (see `kafka` or `nats`).

> is the constructor of processor plugin called once or per message?

No, it's called once on startup. If you want to perform some initialisation before processing every message, then you have to call that logic inside the `Process()` method.
